src/Shared/StringExtensions.cs

```diff
@@ -92,5 +92,10 @@ public static void WriteLine(this TextWriter writer, ReadOnlySpan<char> buffer)
             writer.WriteLine(buffer.ToString());
         }
 #endif
+
+        public static bool IsMSBuildTrueString(this string msbuildString) =>
```

Comment:
I'd add xml docs here that the backing implementation throws if there is a non-bool value. It could surprise someone who used this method as this is a side effect not clear from the name.